core/src/main/java/gov/nist/secauto/metaschema/core/metapath/item/atomic/INumericItem.java

```diff
@@ -78,6 +79,19 @@ static INumericItem cast(@NonNull IAnyAtomicItem item) {
   @NonNull
   BigInteger asInteger();
 
+  // FIXME: Create specialized methods for integer and decimal
+  default int toIntValueExact() {
+    try {
+      return asInteger().intValueExact();
+    } catch (ArithmeticException ex) {
+      throw new CastFunctionException(
+          CastFunctionException.INPUT_VALUE_TOO_LARGE,
+          this,
+          String.format("Input value '%s' is out of range for a Java int.", asString()),
+          ex);
+    }
+  }
```

Comment:
@coderabbitai Please propose specialized implementations for integer and decimal types to address the FIXME comment

Reply:
Thanks. I'll integrate these into the code, but I am going to keep the method in INumericItem without a default body, since the method signature is consistent across the implementations.